src/composables/draw/draw.composable.ts

```diff
@@ -0,0 +1,71 @@
+import { DrawState, useDrawStore } from '@/stores/draw.store'
+import { watch } from 'vue'
+import useDrawInteraction from './draw-interaction.composable'
+import Draw from 'ol/interaction/Draw'
+
+type DrawInteractions = {
+  drawPoint: Draw
+  drawLabel: Draw
+  drawLine: Draw
+  drawCircle: Draw
+  drawPolygon: Draw
+}
+
+export default function useDraw() {
+  const {
+    setDrawPointActive,
+    setDrawLabelActive,
+    setDrawLineActive,
+    setDrawPolygonActive,
+    setDrawCircleActive,
+    drawState,
+  } = useDrawStore()
+  const drawInteractions = {
+    drawPoint: useDrawInteraction({ type: 'Point' }).drawInteraction,
+    drawLabel: useDrawInteraction({ type: 'Point' }).drawInteraction,
+    drawLine: useDrawInteraction({ type: 'LineString' }).drawInteraction,
+    drawCircle: useDrawInteraction({ type: 'Circle' }).drawInteraction,
+    drawPolygon: useDrawInteraction({ type: 'Polygon' }).drawInteraction,
+  } as DrawInteractions
+
+  watch(
+    () => drawState,
+    state => {
+      Object.keys(drawInteractions).forEach(key => {
+        const stateKey = `${key}Active` as keyof DrawState
+        const isActive = state[stateKey]
+        drawInteractions[key as keyof DrawInteractions].setActive(isActive)
```

Comment:
Maybe if we only have one key for the only active tool it could simplify the code? but ok to keep as is if you think this is better.

Reply:
Indeed, this got easier, thanks!